src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/XSV/SimpleKeyXsvFuncotationFactory.java

```diff
@@ -0,0 +1,303 @@
+            populateAnnotationMap( it, permissiveColumns );
+        }
+
+        // Initialize overrides / defaults:
+        initializeAnnotationOverrides( annotationOverrides );
+    }
+
+    //==================================================================================================================
+    // Override Methods:
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public LinkedHashSet<String> getSupportedFuncotationFields() {
+        return new LinkedHashSet<>(annotationColumnNames);
+    }
+
+    @Override
+    /**
+     * {@inheritDoc}
+     * This method should never be called on a {@link SimpleKeyXsvFuncotationFactory} - knowledge of the applied
+     * {@link GencodeFuncotation}s is required to create an {@link XSVFuncotation} from here.
+     *
+     */
+    public List<Funcotation> createFuncotations(final VariantContext variant, final ReferenceContext referenceContext, final List<Feature> featureList) {
+        throw new GATKException("This method should never be called on a " + this.getClass().getName());
+    }
+
+    @Override
+    /**
+     * {@inheritDoc}
+     * For each {@link org.broadinstitute.hellbender.tools.funcotator.dataSources.gencode.GencodeFuncotation}, the Transcript ID or Gene Name (Hugo Symbol)
+     * is checked for a match against the key of any annotation in {@link SimpleKeyXsvFuncotationFactory#annotationMap}.
+     * If a match is found, an {@link XSVFuncotation} is added to the list to be returned.
+     */
+    public List<Funcotation> createFuncotations(final VariantContext variant,
+                                                final ReferenceContext referenceContext,
+                                                final List<Feature> featureList,
+                                                final List<GencodeFuncotation> gencodeFuncotations) {
+        final List<Funcotation> outputFuncotations = new ArrayList<>();
+
+        // If we have gencodeFuncotations we go through them and check for the correct Gene Name / TranscriptID.
+        // If any match, we create our xsvFuncotation for this variant:
+        for (  final GencodeFuncotation gencodeFuncotation : gencodeFuncotations ) {
+
+            // Get our key from the gencode funcotation:
+            final String key;
+            if ( keyType == XsvDataKeyType.GENE_NAME ) {
+                key = gencodeFuncotation.getHugoSymbol();
+            }
+            else {
+                key = gencodeFuncotation.getAnnotationTranscript();
+            }
+
+            // Get our annotations:
+            final List<String> annotations = annotationMap.get( key );
+            if ( annotations != null ) {
+                // Add our annotations to the list:
+                outputFuncotations.add( new XSVFuncotation(annotationColumnNames, annotations) );
+            }
+        }
+
+        setOverrideValuesInFuncotations(outputFuncotations);
```

Comment:
The caller currently has no notion of what the names of the Funcotations are that each factory creates, so it can't override them.  That is, a Funcotation is not a sorted map of funcotationName->value, it's just an object that knows how to serialize itself.  Because of this, each Funcotation factory must set its own overrides for the fields it owns so that when each Funcotation is created the overrides can be applied.

For the fields not owned by any FuncotationFactory, Funcotator will add in an annotation as a default value (it does this by creating a list of unaccounted for override values as part of `onTraversalStart`.